code/src/components/Form.js

```diff
@@ -0,0 +1,296 @@
+import React, {useState} from 'react'
+
+import questions from '../questions.json'
+import { Input } from 'components/Input'
+import { Select } from 'components/Select'
+import { Button } from 'components/Button'
+import { RadioButton } from 'components/RadioButton'
+
+export const Form = () => {
+	const [page, setPage] = useState(0);
+	const [userName, setUserName] = useState('');
+	const [breakupPerson, setBreakupPerson] = useState('');
+	const [shipDuration, setShipDuration] = useState('default');
+	const [breakupReason, setBreakupReason] = useState('');
+	const [traitsCheckboxes, setTraitsCheckboxes] = useState([])
+	const [getBackTogether, setGetBackTogether] = useState('')
+	const [dateFriend, setDateFriend] = useState('')
+	const [friendContactDetails, setFriendContactDetails] = useState('')
+	const [error, setError] = useState('')
+
+	const onSubmitting = (e) => {
+		e.preventDefault();
+	};
+
+	const handleUserNameChange = (event) => {
+		setUserName(event.target.value)
+	}
+
+	const handleBreakupReason = (event) => {
+		setBreakupReason(event.target.value)
+	}
+
+	const handleDateFriend = (event) => {
+		setDateFriend(event.target.value)
+	}
+	
+	const handleGetBackTogether = (event) => {
+		setGetBackTogether(event.target.value)
+	}
+
+	const handleFriendContactDetails = (event) => {
+		setFriendContactDetails(event.target.value)
+	}

```

Comment:
Smart putting all of these in the Form app... I just used anonymous in-line functions, but I can see for re-usability of components that this is a much better way of doing things!